### PR TITLE
ICSAAS-364 refinement of configured filters logic

### PR DESCRIPTION
**Pull request by igorsimovski**
- filter entries should be 'AND' related
 - a filter entry with multiple comma separated values is considered a composite filter of 'OR' related filter entries
 - create a helper class for parsing the configured filters
  - add unit test
 - update filters dialog field definition

### CODE REVIEW

File: core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/ConfiguredFiltersParser.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+
+@RequiredArgsConstructor
+public class ConfiguredFiltersParser {
```
Comment (nhirrle) [inline core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/ConfiguredFiltersParser.java:16]: why don't we use a traditional sling model for the parsing?

Comment (igorsimovski) [reply core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/ConfiguredFiltersParser.java]: I moved the parsing/mapping logic in the FilterModelmpl sling model.
Maybe one reason why a helper class seems ok to me, is the light weight unit test (avoid usage of aem mocks)

File: ui.apps/src/main/content/jcr_root/apps/saas-aem-module/components/search/_cq_dialog.xml [hunk 2]
```diff
@@ -41,10 +41,6 @@
                                                 fieldLabel="Language"
                                                 fieldDescription="Optional override of context resolved language."
                                                 name="./language"/>
-                                            <filters
-                                                jcr:primaryType="nt:unstructured"
-                                                sling:resourceType="granite/ui/components/coral/foundation/include"
-                                                path="/apps/saas-aem-module/shared/dialog/filters"/>
                                             <resultsPerPage
```
Comment (Valtech-Switzerland) [inline ui.apps/src/main/content/jcr_root/apps/saas-aem-module/components/search/_cq_dialog.xml:44]: - [ ] move this at the end of the tab

File: core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/FilterModelImpl.java [hunk 3]
```diff
@@ -23,4 +32,27 @@
   @ValueMapValue
   private String value;
 
+  @Getter
+  private Filter filter;
+
+  @PostConstruct
+  private void init() {
```
Comment (Valtech-Switzerland) [inline core/src/main/java/com/valtech/aem/saas/core/fulltextsearch/FilterModelImpl.java:39]: - [ ] please check if this is unit tested

**Review by Valtech-Switzerland — APPROVED** [review]

**Review by nhirrle — APPROVED** [review]

**Pull request merged**